Allow to remove the Game status bar in GameWindow + comments

Nesqwik opened the pull request:
This changes allow to create a game window without the status bar.
I also added comments to explaine some functions.

DamienCassou commented on src/main/java/gameframework/gui/GameWindow.java:
```diff
@@ -13,30 +14,83 @@
 	protected final GameStatusBar statusBar = new GameStatusBar();
+
+
+	@Deprecated
```
why do you add a deprecated constructor?

DamienCassou commented on src/main/java/gameframework/gui/GameWindow.java:
```diff
@@ -13,30 +14,83 @@
 
 public class GameWindow {
 
-	protected final Frame frame;
```
I don't like that. Can you refactor your code to avoid this change?

DamienCassou commented:
Instead of adding all these constructors, can't you add `showStatusBar()` and `hideStatusBar()` instead?

Nesqwik commented:
I have changed a lot of stuff to handle an "home-made" status bar :
 - I'ts now possible to create a custom status bar. The data can now be stored into "gameData". This allow to create as many statusBarElement as you need, not only score and life. 
 - I added the "showStatusBar()" and "hideStatusBar()" method.

About the "@Deprecated" constructor, it was already there when i work on this changes. It had to be removeD from the master branch just before my pull request. I deleted this on my last commit.

DamienCassou commented on src/main/java/gameframework/game/GameData.java:
```diff
@@ -8,11 +8,17 @@
+	 * Map of observable values to save any needed data.
+	 */
+	protected final Map<String, ObservableValue<?>> observableValues;
 	protected final ObservableValue<Integer> score;
```
do we still need `score` and `life`?

DamienCassou commented on src/main/java/gameframework/gui/GameWindow.java:
```diff
@@ -32,11 +62,39 @@ public GameWindow(String gameName, GameCanvas gameCanvas,
 		this.statusBar.addAll(elementsStatusBar);
 		this.frame = new Frame(gameName);
 		this.gameCanvas = gameCanvas;
-		this.gameCanvas.setSize(//
```
please don't touch lines if you don't have anything to change. Keep the diff as simple as possible to ease code review

DamienCassou commented on src/main/java/gameframework/gui/GameWindow.java:
```diff
@@ -32,11 +62,39 @@ public GameWindow(String gameName, GameCanvas gameCanvas,
 	}
+	
+	/**
+	 * hide the status bar
```
remove useless comments

DamienCassou commented on src/main/java/gameframework/gui/GameWindow.java:
```diff
@@ -32,11 +62,39 @@ public GameWindow(String gameName, GameCanvas gameCanvas,
+	 * Shows or hide the status bar depending of the value of parameter showStatusBar.
+	 * @param showStatusBar if true - shows the status bar; otherwise, hides the status bar.
+	 */
+	private void showStatusBar(boolean showStatusBar) {
```
please replace `private` by `protected` everywhere, you never know what your subclasses will need to do.

DamienCassou replied:
what about `toggleStatusBarVisibility` instead of `showStatusBar`?

DamienCassou commented on src/main/java/gameframework/gui/GameWindow.java:
```diff
@@ -53,9 +111,13 @@ public void windowClosing(WindowEvent e) {
 	}
 
+	/**
+	 * @param comp
```
useless

DamienCassou commented on src/main/java/gameframework/gui/GameWindow.java:
```diff
@@ -53,9 +111,13 @@ public void windowClosing(WindowEvent e) {
 	}
 
-}
+}
```
please always keep the empty line at the end of all files

DamienCassou commented:
Please take care of the remaining comment and squash your commits into 1. I will then merge your PR. Thanks

DamienCassou commented:
No updates for 14 days, I'm closing this.
